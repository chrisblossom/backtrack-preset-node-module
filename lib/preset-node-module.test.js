'use strict';

const transformConfig = require('@backtrack/core/dist/options-file/transform-config')
    .transformConfig;

test('preset returns expected config - defaults flow: true', () => {
    const backtrackConfig = {
        presets: ['../'],
    };

    const result = transformConfig(backtrackConfig, __dirname);

    expect(result).toMatchSnapshot();
});

test('adds flow with flow: true option', () => {
    const backtrackConfig = {
        presets: [['../', { flow: true }]],
    };

    const result = transformConfig(backtrackConfig, __dirname);

    expect(result).toMatchSnapshot();
});
