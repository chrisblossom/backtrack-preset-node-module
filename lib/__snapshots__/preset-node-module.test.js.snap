// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset returns expected config - defaults flow: true 1`] = `
Object {
  "build": Array [
    "backtrack clean",
    Array [
      Object {
        "name": "babel",
        "task": "babel src -d dist --ignore test.js,__sandbox__/ --source-maps",
      },
      Object {
        "name": "flow-copy-source",
        "task": "flow-copy-source --ignore \\"**/*.test.js\\" --ignore \\"**/__sandbox__/**/*\\" src dist",
      },
    ],
  ],
  "check-all": Array [
    Array [
      "backtrack lint",
      "backtrack test",
      "backtrack flow",
    ],
  ],
  "clean": Object {
    "copy": Array [],
    "del": Array [
      "**/*",
    ],
    "makeDirs": Array [],
  },
  "dev": Array [
    "backtrack clean",
    Array [
      Object {
        "name": "babel",
        "task": "babel -w src/ -d dist/ --ignore test.js,__sandbox__/ --source-maps",
      },
      Object {
        "name": "flow-copy-source",
        "task": "flow-copy-source -w --ignore \\"**/*.test.js\\" --ignore \\"**/__sandbox__/**/*\\" src dist",
      },
    ],
  ],
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".babelrc.js": "<PROJECT_ROOT>/.babelrc.js",
        ".circleci/config.yml": "<PROJECT_ROOT>/.circleci/config.yml",
        ".editorconfig": "<PROJECT_ROOT>/.editorconfig",
        ".eslintignore": "<PROJECT_ROOT>/.eslintignore",
        ".eslintrc.js": "<PROJECT_ROOT>/.eslintrc.js",
        ".flowconfig": "<PROJECT_ROOT>/.flowconfig",
        ".gitignore": "<PROJECT_ROOT>/.gitignore",
        ".npmrc": "<PROJECT_ROOT>/.npmrc",
        ".prettierignore": "<PROJECT_ROOT>/.prettierignore",
        ".prettierrc.js": "<PROJECT_ROOT>/.prettierrc.js",
        ".yarnrc": "<PROJECT_ROOT>/.yarnrc",
        "CHANGELOG.md": "<PROJECT_ROOT>/CHANGELOG.md",
        "README.md": "<PROJECT_ROOT>/README.md",
        "appveyor.yml": "<PROJECT_ROOT>/appveyor.yml",
        "jest.config.js": "<PROJECT_ROOT>/jest.config.js",
        "src/preset-node-module.js": "<PROJECT_ROOT>/src/preset-node-module.js",
        "wallaby.config.js": "<PROJECT_ROOT>/wallaby.config.js",
      },
      "allowChanges": Object {
        ".babelrc.js": false,
        ".circleci/config.yml": false,
        ".editorconfig": false,
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".flowconfig": false,
        ".gitignore": false,
        ".npmrc": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
        ".yarnrc": false,
        "CHANGELOG.md": true,
        "README.md": true,
        "appveyor.yml": false,
        "jest.config.js": false,
        "src/preset-node-module.js": true,
        "wallaby.config.js": false,
      },
      "files": Array [
        ".eslintrc.js",
        ".eslintignore",
        ".prettierrc.js",
        ".prettierignore",
        "jest.config.js",
        "wallaby.config.js",
        ".circleci/config.yml",
        "appveyor.yml",
        "CHANGELOG.md",
        "README.md",
        ".editorconfig",
        ".gitignore",
        ".npmrc",
        ".yarnrc",
        "src/preset-node-module.js",
        ".babelrc.js",
        ".flowconfig",
      ],
      "hash": Object {
        ".babelrc.js": "",
        ".circleci/config.yml": "3b7696d7b955ebd4ac0f5eeacfff227acea15301b6e450b111677d32411b2a8b",
        ".editorconfig": "65c8c6dd2b55cd63f7a55d2ed77bb1344a345ed651c5b0a1cb689fa95753e63d",
        ".eslintignore": "7b358a16955be6b9b56add38102fccd293af540701ac91cd5020bfe37fe3fbb3",
        ".eslintrc.js": "0de8c62f28870b2afeb3eb58a5a9a106a96c9181aa9478f71760bb155bb4a205",
        ".flowconfig": "",
        ".gitignore": "c204ca8726ae7f7e0aedcd854a49064fd5a2385de43318f58e8cd70e7dd5adbd",
        ".npmrc": "f92bfc9e4d622a5c53c3fcd1b61206420f8dc666db99e57560c33ad2fdc80a47",
        ".prettierignore": "7b358a16955be6b9b56add38102fccd293af540701ac91cd5020bfe37fe3fbb3",
        ".prettierrc.js": "d30f794e5447d44ad2ed8a1123bc15fdaf6775ad7776f3c1411e3b965f3d3a20",
        ".yarnrc": "e4bdd4d0bcb6dd15e5585dc6fa2c90780abaf4904305afb12d9af02000e070f3",
        "CHANGELOG.md": "f55181dcff3c688e1b0b62f38b6e3187de326c22b4cc1eca3642bf63d6ea2552",
        "README.md": "d7d76c773beb0046387163769d9cd56dee330085585c83637c4e3e8f6757d348",
        "appveyor.yml": "e26fc69a9c740e6fe6267ea2da435aa54c83a110eface66642470dc2e8336713",
        "jest.config.js": "2dfc22f99102a3d4d2335ad31b24e0902a3b81348aa74acf270d9c88ed0a2bf3",
        "src/preset-node-module.js": "",
        "wallaby.config.js": "b79a89efd0484c2aba8477197056e4ae331dc6a0d533e050edd99d89630cb78f",
      },
      "ignoreUpdates": Object {
        ".babelrc.js": false,
        ".circleci/config.yml": false,
        ".editorconfig": false,
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".flowconfig": false,
        ".gitignore": false,
        ".npmrc": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
        ".yarnrc": false,
        "CHANGELOG.md": true,
        "README.md": true,
        "appveyor.yml": false,
        "jest.config.js": false,
        "src/preset-node-module.js": true,
        "wallaby.config.js": false,
      },
    },
    "makeDirs": Array [
      ".circleci",
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/CHANGELOG.md": "<PROJECT_ROOT>/lib/files/CHANGELOG.md",
        "lib/files/README.md": "<PROJECT_ROOT>/lib/files/README.md",
        "lib/files/babelrc-flow.js": "<PROJECT_ROOT>/lib/files/babelrc-flow.js",
        "lib/files/editorconfig": "<PROJECT_ROOT>/lib/files/editorconfig",
        "lib/files/flowconfig": "<PROJECT_ROOT>/lib/files/flowconfig",
        "lib/files/gitignore": "<PROJECT_ROOT>/lib/files/gitignore",
        "lib/files/npmrc": "<PROJECT_ROOT>/lib/files/npmrc",
        "lib/files/package-entry-flow.js": "<PROJECT_ROOT>/lib/files/package-entry-flow.js",
        "lib/files/yarnrc": "<PROJECT_ROOT>/lib/files/yarnrc",
        "node_modules/@backtrack/preset-jest/lib/files/appveyor.yml": "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib/files/appveyor.yml",
        "node_modules/@backtrack/preset-jest/lib/files/circleci-v2.yml": "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib/files/circleci-v2.yml",
        "node_modules/@backtrack/preset-jest/lib/files/jest.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib/files/jest.js",
        "node_modules/@backtrack/preset-jest/lib/files/wallaby.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib/files/wallaby.js",
        "node_modules/@backtrack/preset-style/lib/files/eslint.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-style/lib/files/eslint.js",
        "node_modules/@backtrack/preset-style/lib/files/ignore": "<PROJECT_ROOT>/node_modules/@backtrack/preset-style/lib/files/ignore",
        "node_modules/@backtrack/preset-style/lib/files/prettier.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-style/lib/files/prettier.js",
      },
      "files": Array [
        "node_modules/@backtrack/preset-style/lib/files/eslint.js",
        "node_modules/@backtrack/preset-style/lib/files/ignore",
        "node_modules/@backtrack/preset-style/lib/files/prettier.js",
        "node_modules/@backtrack/preset-style/lib/files/ignore",
        "node_modules/@backtrack/preset-jest/lib/files/jest.js",
        "node_modules/@backtrack/preset-jest/lib/files/wallaby.js",
        "node_modules/@backtrack/preset-jest/lib/files/circleci-v2.yml",
        "node_modules/@backtrack/preset-jest/lib/files/appveyor.yml",
        "lib/files/CHANGELOG.md",
        "lib/files/README.md",
        "lib/files/editorconfig",
        "lib/files/gitignore",
        "lib/files/npmrc",
        "lib/files/yarnrc",
        "lib/files/package-entry-flow.js",
        "lib/files/babelrc-flow.js",
        "lib/files/flowconfig",
      ],
      "hash": Object {
        "lib/files/CHANGELOG.md": "ignoreUpdates",
        "lib/files/README.md": "ignoreUpdates",
        "lib/files/babelrc-flow.js": "31340d9716a55146d1d2f2573bb6b4a67c4957774fb1c7175397966be7b0e10f",
        "lib/files/editorconfig": "65c8c6dd2b55cd63f7a55d2ed77bb1344a345ed651c5b0a1cb689fa95753e63d",
        "lib/files/flowconfig": "a097b005abea75e2e62d08a1cdbf12263aa9e5c220be921f7992a1cbc1e1bd18",
        "lib/files/gitignore": "715c4f869d50d4e11254a97441df027e6cabd50612d10e844e5efba4078f484f",
        "lib/files/npmrc": "f92bfc9e4d622a5c53c3fcd1b61206420f8dc666db99e57560c33ad2fdc80a47",
        "lib/files/package-entry-flow.js": "ignoreUpdates",
        "lib/files/yarnrc": "e4bdd4d0bcb6dd15e5585dc6fa2c90780abaf4904305afb12d9af02000e070f3",
        "node_modules/@backtrack/preset-jest/lib/files/appveyor.yml": "e26fc69a9c740e6fe6267ea2da435aa54c83a110eface66642470dc2e8336713",
        "node_modules/@backtrack/preset-jest/lib/files/circleci-v2.yml": "3b7696d7b955ebd4ac0f5eeacfff227acea15301b6e450b111677d32411b2a8b",
        "node_modules/@backtrack/preset-jest/lib/files/jest.js": "2dfc22f99102a3d4d2335ad31b24e0902a3b81348aa74acf270d9c88ed0a2bf3",
        "node_modules/@backtrack/preset-jest/lib/files/wallaby.js": "d37d10441a299a78a0b6b608fd44add196354f55a6eee2e9854cfef49baa033f",
        "node_modules/@backtrack/preset-style/lib/files/eslint.js": "0de8c62f28870b2afeb3eb58a5a9a106a96c9181aa9478f71760bb155bb4a205",
        "node_modules/@backtrack/preset-style/lib/files/ignore": "7b358a16955be6b9b56add38102fccd293af540701ac91cd5020bfe37fe3fbb3",
        "node_modules/@backtrack/preset-style/lib/files/prettier.js": "d30f794e5447d44ad2ed8a1123bc15fdaf6775ad7776f3c1411e3b965f3d3a20",
      },
    },
  },
  "flow": Array [
    "flow",
  ],
  "format": Array [
    "prettier **/*.{js,md,json} --write",
  ],
  "lint": Array [
    "eslint .",
  ],
  "lint.fix": Array [
    "eslint --fix .",
  ],
  "packageJson": Array [
    Object {
      "lint-staged": Object {
        "*.{js,md,json}": Array [
          "prettier --write",
          "git add",
        ],
      },
    },
    Object {
      "scripts": Object {
        "test.update": "jest --updateSnapshot",
        "test.watch": "jest --watch",
      },
    },
    Object {
      "babel": Object {
        "presets": Array [
          "./.babelrc.js",
        ],
      },
      "devEngines": Object {
        "node": ">=8.9.0",
        "npm": ">=5.5.1",
      },
      "engines": Object {
        "node": ">=6.9.0",
        "npm": ">=3.10.10",
        "yarn": ">=1.6.0",
      },
      "files": Array [
        "dist/",
        "flow-typed/",
      ],
      "main": "dist/preset-node-module.js",
    },
  ],
  "precommit": Array [
    "lint-staged",
  ],
  "prepublishOnly": Array [
    "backtrack check-all",
    "backtrack build",
  ],
  "prepush": Array [
    "backtrack check-all",
  ],
  "resolve": Object {
    "@backtrack/preset-jest": "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib",
    "@backtrack/preset-node-module": "<PROJECT_ROOT>/lib",
    "@backtrack/preset-style": "<PROJECT_ROOT>/node_modules/@backtrack/preset-style/lib",
  },
  "test": Array [
    "jest --passWithNoTests",
  ],
  "test.ci": Array [
    "jest --passWithNoTests --coverage",
    "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib/send-coverage.js",
  ],
  "test.ci-pretest": Array [
    [Function noop],
    "backtrack build",
    "backtrack lint",
    "<PROJECT_ROOT>/lib/ci-run-flow.js",
  ],
}
`;

exports[`removes flow with flow: false option 1`] = `
Object {
  "build": Array [
    "backtrack clean",
    Array [
      Object {
        "name": "babel",
        "task": "babel src -d dist --ignore test.js,__sandbox__/ --source-maps",
      },
    ],
  ],
  "check-all": Array [
    Array [
      "backtrack lint",
      "backtrack test",
    ],
  ],
  "clean": Object {
    "copy": Array [],
    "del": Array [
      "**/*",
    ],
    "makeDirs": Array [],
  },
  "dev": Array [
    "backtrack clean",
    Array [
      Object {
        "name": "babel",
        "task": "babel -w src/ -d dist/ --ignore test.js,__sandbox__/ --source-maps",
      },
    ],
  ],
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".babelrc.js": "<PROJECT_ROOT>/.babelrc.js",
        ".circleci/config.yml": "<PROJECT_ROOT>/.circleci/config.yml",
        ".editorconfig": "<PROJECT_ROOT>/.editorconfig",
        ".eslintignore": "<PROJECT_ROOT>/.eslintignore",
        ".eslintrc.js": "<PROJECT_ROOT>/.eslintrc.js",
        ".gitignore": "<PROJECT_ROOT>/.gitignore",
        ".npmrc": "<PROJECT_ROOT>/.npmrc",
        ".prettierignore": "<PROJECT_ROOT>/.prettierignore",
        ".prettierrc.js": "<PROJECT_ROOT>/.prettierrc.js",
        ".yarnrc": "<PROJECT_ROOT>/.yarnrc",
        "CHANGELOG.md": "<PROJECT_ROOT>/CHANGELOG.md",
        "README.md": "<PROJECT_ROOT>/README.md",
        "appveyor.yml": "<PROJECT_ROOT>/appveyor.yml",
        "jest.config.js": "<PROJECT_ROOT>/jest.config.js",
        "src/preset-node-module.js": "<PROJECT_ROOT>/src/preset-node-module.js",
        "wallaby.config.js": "<PROJECT_ROOT>/wallaby.config.js",
      },
      "allowChanges": Object {
        ".babelrc.js": false,
        ".circleci/config.yml": false,
        ".editorconfig": false,
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".gitignore": false,
        ".npmrc": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
        ".yarnrc": false,
        "CHANGELOG.md": true,
        "README.md": true,
        "appveyor.yml": false,
        "jest.config.js": false,
        "src/preset-node-module.js": true,
        "wallaby.config.js": false,
      },
      "files": Array [
        ".eslintrc.js",
        ".eslintignore",
        ".prettierrc.js",
        ".prettierignore",
        "jest.config.js",
        "wallaby.config.js",
        ".circleci/config.yml",
        "appveyor.yml",
        "CHANGELOG.md",
        "README.md",
        ".editorconfig",
        ".gitignore",
        ".npmrc",
        ".yarnrc",
        "src/preset-node-module.js",
        ".babelrc.js",
      ],
      "hash": Object {
        ".babelrc.js": "",
        ".circleci/config.yml": "3b7696d7b955ebd4ac0f5eeacfff227acea15301b6e450b111677d32411b2a8b",
        ".editorconfig": "65c8c6dd2b55cd63f7a55d2ed77bb1344a345ed651c5b0a1cb689fa95753e63d",
        ".eslintignore": "7b358a16955be6b9b56add38102fccd293af540701ac91cd5020bfe37fe3fbb3",
        ".eslintrc.js": "0de8c62f28870b2afeb3eb58a5a9a106a96c9181aa9478f71760bb155bb4a205",
        ".gitignore": "c204ca8726ae7f7e0aedcd854a49064fd5a2385de43318f58e8cd70e7dd5adbd",
        ".npmrc": "f92bfc9e4d622a5c53c3fcd1b61206420f8dc666db99e57560c33ad2fdc80a47",
        ".prettierignore": "7b358a16955be6b9b56add38102fccd293af540701ac91cd5020bfe37fe3fbb3",
        ".prettierrc.js": "d30f794e5447d44ad2ed8a1123bc15fdaf6775ad7776f3c1411e3b965f3d3a20",
        ".yarnrc": "e4bdd4d0bcb6dd15e5585dc6fa2c90780abaf4904305afb12d9af02000e070f3",
        "CHANGELOG.md": "f55181dcff3c688e1b0b62f38b6e3187de326c22b4cc1eca3642bf63d6ea2552",
        "README.md": "d7d76c773beb0046387163769d9cd56dee330085585c83637c4e3e8f6757d348",
        "appveyor.yml": "e26fc69a9c740e6fe6267ea2da435aa54c83a110eface66642470dc2e8336713",
        "jest.config.js": "2dfc22f99102a3d4d2335ad31b24e0902a3b81348aa74acf270d9c88ed0a2bf3",
        "src/preset-node-module.js": "",
        "wallaby.config.js": "b79a89efd0484c2aba8477197056e4ae331dc6a0d533e050edd99d89630cb78f",
      },
      "ignoreUpdates": Object {
        ".babelrc.js": false,
        ".circleci/config.yml": false,
        ".editorconfig": false,
        ".eslintignore": false,
        ".eslintrc.js": false,
        ".gitignore": false,
        ".npmrc": false,
        ".prettierignore": false,
        ".prettierrc.js": false,
        ".yarnrc": false,
        "CHANGELOG.md": true,
        "README.md": true,
        "appveyor.yml": false,
        "jest.config.js": false,
        "src/preset-node-module.js": true,
        "wallaby.config.js": false,
      },
    },
    "makeDirs": Array [
      ".circleci",
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/CHANGELOG.md": "<PROJECT_ROOT>/lib/files/CHANGELOG.md",
        "lib/files/README.md": "<PROJECT_ROOT>/lib/files/README.md",
        "lib/files/babelrc.js": "<PROJECT_ROOT>/lib/files/babelrc.js",
        "lib/files/editorconfig": "<PROJECT_ROOT>/lib/files/editorconfig",
        "lib/files/gitignore": "<PROJECT_ROOT>/lib/files/gitignore",
        "lib/files/npmrc": "<PROJECT_ROOT>/lib/files/npmrc",
        "lib/files/package-entry.js": "<PROJECT_ROOT>/lib/files/package-entry.js",
        "lib/files/yarnrc": "<PROJECT_ROOT>/lib/files/yarnrc",
        "node_modules/@backtrack/preset-jest/lib/files/appveyor.yml": "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib/files/appveyor.yml",
        "node_modules/@backtrack/preset-jest/lib/files/circleci-v2.yml": "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib/files/circleci-v2.yml",
        "node_modules/@backtrack/preset-jest/lib/files/jest.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib/files/jest.js",
        "node_modules/@backtrack/preset-jest/lib/files/wallaby.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib/files/wallaby.js",
        "node_modules/@backtrack/preset-style/lib/files/eslint.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-style/lib/files/eslint.js",
        "node_modules/@backtrack/preset-style/lib/files/ignore": "<PROJECT_ROOT>/node_modules/@backtrack/preset-style/lib/files/ignore",
        "node_modules/@backtrack/preset-style/lib/files/prettier.js": "<PROJECT_ROOT>/node_modules/@backtrack/preset-style/lib/files/prettier.js",
      },
      "files": Array [
        "node_modules/@backtrack/preset-style/lib/files/eslint.js",
        "node_modules/@backtrack/preset-style/lib/files/ignore",
        "node_modules/@backtrack/preset-style/lib/files/prettier.js",
        "node_modules/@backtrack/preset-style/lib/files/ignore",
        "node_modules/@backtrack/preset-jest/lib/files/jest.js",
        "node_modules/@backtrack/preset-jest/lib/files/wallaby.js",
        "node_modules/@backtrack/preset-jest/lib/files/circleci-v2.yml",
        "node_modules/@backtrack/preset-jest/lib/files/appveyor.yml",
        "lib/files/CHANGELOG.md",
        "lib/files/README.md",
        "lib/files/editorconfig",
        "lib/files/gitignore",
        "lib/files/npmrc",
        "lib/files/yarnrc",
        "lib/files/package-entry.js",
        "lib/files/babelrc.js",
      ],
      "hash": Object {
        "lib/files/CHANGELOG.md": "ignoreUpdates",
        "lib/files/README.md": "ignoreUpdates",
        "lib/files/babelrc.js": "2ba1900d7d5148a5834dc7863dbc26ff75e03dcf503c1ba973361e865fe6f6d3",
        "lib/files/editorconfig": "65c8c6dd2b55cd63f7a55d2ed77bb1344a345ed651c5b0a1cb689fa95753e63d",
        "lib/files/gitignore": "715c4f869d50d4e11254a97441df027e6cabd50612d10e844e5efba4078f484f",
        "lib/files/npmrc": "f92bfc9e4d622a5c53c3fcd1b61206420f8dc666db99e57560c33ad2fdc80a47",
        "lib/files/package-entry.js": "ignoreUpdates",
        "lib/files/yarnrc": "e4bdd4d0bcb6dd15e5585dc6fa2c90780abaf4904305afb12d9af02000e070f3",
        "node_modules/@backtrack/preset-jest/lib/files/appveyor.yml": "e26fc69a9c740e6fe6267ea2da435aa54c83a110eface66642470dc2e8336713",
        "node_modules/@backtrack/preset-jest/lib/files/circleci-v2.yml": "3b7696d7b955ebd4ac0f5eeacfff227acea15301b6e450b111677d32411b2a8b",
        "node_modules/@backtrack/preset-jest/lib/files/jest.js": "2dfc22f99102a3d4d2335ad31b24e0902a3b81348aa74acf270d9c88ed0a2bf3",
        "node_modules/@backtrack/preset-jest/lib/files/wallaby.js": "d37d10441a299a78a0b6b608fd44add196354f55a6eee2e9854cfef49baa033f",
        "node_modules/@backtrack/preset-style/lib/files/eslint.js": "0de8c62f28870b2afeb3eb58a5a9a106a96c9181aa9478f71760bb155bb4a205",
        "node_modules/@backtrack/preset-style/lib/files/ignore": "7b358a16955be6b9b56add38102fccd293af540701ac91cd5020bfe37fe3fbb3",
        "node_modules/@backtrack/preset-style/lib/files/prettier.js": "d30f794e5447d44ad2ed8a1123bc15fdaf6775ad7776f3c1411e3b965f3d3a20",
      },
    },
  },
  "format": Array [
    "prettier **/*.{js,md,json} --write",
  ],
  "lint": Array [
    "eslint .",
  ],
  "lint.fix": Array [
    "eslint --fix .",
  ],
  "packageJson": Array [
    Object {
      "lint-staged": Object {
        "*.{js,md,json}": Array [
          "prettier --write",
          "git add",
        ],
      },
    },
    Object {
      "scripts": Object {
        "test.update": "jest --updateSnapshot",
        "test.watch": "jest --watch",
      },
    },
    Object {
      "babel": Object {
        "presets": Array [
          "./.babelrc.js",
        ],
      },
      "devEngines": Object {
        "node": ">=8.9.0",
        "npm": ">=5.5.1",
      },
      "engines": Object {
        "node": ">=6.9.0",
        "npm": ">=3.10.10",
        "yarn": ">=1.6.0",
      },
      "files": Array [
        "dist/",
      ],
      "main": "dist/preset-node-module.js",
    },
  ],
  "precommit": Array [
    "lint-staged",
  ],
  "prepublishOnly": Array [
    "backtrack check-all",
    "backtrack build",
  ],
  "prepush": Array [
    "backtrack check-all",
  ],
  "resolve": Object {
    "@backtrack/preset-jest": "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib",
    "@backtrack/preset-node-module": "<PROJECT_ROOT>/lib",
    "@backtrack/preset-style": "<PROJECT_ROOT>/node_modules/@backtrack/preset-style/lib",
  },
  "test": Array [
    "jest --passWithNoTests",
  ],
  "test.ci": Array [
    "jest --passWithNoTests --coverage",
    "<PROJECT_ROOT>/node_modules/@backtrack/preset-jest/lib/send-coverage.js",
  ],
  "test.ci-pretest": Array [
    [Function noop],
    "backtrack build",
    "backtrack lint",
  ],
}
`;
